### DISCUSSION TITLE
Problem when missing original files from tar_files_input()

### DISCUSSION THREAD

**Discussion by JVAQUEROM** [ropensci/targets#741] [Help] [answered]
Hi,

I want to use targets with a heavy load of data. Therefore, when my data is already read and processed, sometimes is in my interest to move these data files to another external drive and remove then from the work station. To avoid everything re-running, I tried `tar_cue(mode="never")` or `tar_cue(file=FALSE)`, according to @wlandau suggestion. However I get an error "Error : files must have length > 0. ". I'll post a reproducible example below:

```
library(targets)


     tar_dir({ # tar_dir() runs code from a temporary directory.
     tar_script({
library(tarchetypes)
write.table(mtcars, file = "/mnt/data/mtcars.txt") # You can change the directory, remember to change it everywhere in the script
       list(tar_files_input(mtcars_file, list.files("/mnt/data/", pattern=".txt", full.names = TRUE)), tar_target(mtcars_data, read.table(mtcars_file)))
     })
     tar_make() # This builds everything
     file.remove("/mnt/data/mtcars.txt") # Now, I remove the original file
     tar_script({
library(tarchetypes)
       list(tar_files_input(mtcars_file, list.files("/mnt/data/", pattern=".txt", full.names = TRUE), cue = tar_cue(mode="never")), tar_target(mtcars_data, read.table(mtcars_file)))
     })
     tar_make() # This will error
     tar_script({
library(tarchetypes)
       list(tar_files_input(mtcars_file, list.files("/mnt/data/", pattern=".txt", full.names=TRUE), cue = tar_cue(file=FALSE)), tar_target(mtcars_data, read.table(mtcars_file)))
     })
     tar_make() # This will error again
     })
```

Could this be a bug? Any help is welcome.

**Comment by wlandau** [comment]
This is part of the design of `tarchetypes::tar_files_input()` specifically, it statically checks the character vector of files when it is defining the targets. In other words, the `files` argument of `tar_files_input()` is handled differently than the `command` argument of other target factories like `tar_files()`.

```r
> tarchetypes::tar_files_input(a, files = character(0)) # takes `files`, not `command`
Error: files must have length > 0.
Run `rlang::last_error()` to see where the error occurred.
```

If you want `tar_files_input()` to work when you delete the files, you will need to supply the original character vector of file path names instead of writing `list.files()` because the output of `list.files()` changes once you delete those files. Or you could choose a different target factory instead of `tar_files_input()`.

Somewhat related: #728.

**Reply by JVAQUEROM** [reply]
Thank you for your quick answer. This is indeed interesting. I guess I can set the filelist (there are hundreds) as a global variable before the targets list in the _targets.R file. I'll see if that works.

**Reply by JVAQUEROM** [reply, marked as answer]
I created a modified list.files function 

```
mylist <- function(x, ...){
    y <- list.files(x, ...)
    if(length(y) == 0) {
	y <- NA_character_
    }
    y
}
```
This one works fine with the `tar_cue(mode = "never")` option.


Thanks, very interesting ideed
